src/test/java/org/opensearch/neuralsearch/processor/combination/GeometricMeanScoreCombinationTechniqueTests.java

```diff
@@ -0,0 +1,98 @@
+        testLogic_whenAllScoresPresentAndNoWeights_thenCorrectScores(technique);
+    }
+
+    public void testLogic_whenNotAllScoresPresentAndNoWeights_thenCorrectScores() {
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(Map.of(), scoreCombinationUtil);
+        testLogic_whenNotAllScoresPresentAndNoWeights_thenCorrectScores(technique);
+    }
+
+    public void testLogic_whenAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Float> scores = List.of(1.0f, 0.5f, 0.3f);
+        List<Double> weights = List.of(0.9, 0.2, 0.7);
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        float expectedScore = 0.5797f;
+        testLogic_whenAllScoresAndWeightsPresent_thenCorrectScores(technique, scores, expectedScore);
+    }
+
+    public void testRandomValues_whenAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Double> weights = IntStream.range(0, RANDOM_SCORES_SIZE)
+            .mapToObj(i -> RandomizedTest.randomDouble())
+            .collect(Collectors.toList());
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        testRandomValues_whenAllScoresAndWeightsPresent_thenCorrectScores(technique, weights);
+    }
+
+    public void testLogic_whenNotAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Float> scores = List.of(1.0f, -1.0f, 0.6f);
+        List<Double> weights = List.of(0.9, 0.2, 0.7);
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        float expectedScore = 0.7997f;
+        testLogic_whenNotAllScoresAndWeightsPresent_thenCorrectScores(technique, scores, expectedScore);
+    }
+
+    public void testRandomValues_whenNotAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Double> weights = IntStream.range(0, RANDOM_SCORES_SIZE)
+            .mapToObj(i -> RandomizedTest.randomDouble())
+            .collect(Collectors.toList());
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        testRandomValues_whenNotAllScoresAndWeightsPresent_thenCorrectScores(technique, weights);
+    }
+
+    /**
+     * Verify score correctness by using alternative formula for geometric mean as n-th root of product of weighted scores,
+     * more details in here https://en.wikipedia.org/wiki/Weighted_geometric_mean
+     */
+    private float geometricMean(List<Float> scores, List<Double> weights) {
```

Comment:
I still have a doubt on the effectiveness of this test code. l believe we don’t need test code based on random number. I would like to here other opinions though.

Reply:
I don't have any concern as long long as test is able to fail if there are changes in formula.